### PR TITLE
Adadelta formula fixes

### PR DESCRIPTION
**Pull request by goldmermaid**
Formula fixes based on https://arxiv.org/abs/1212.5701.

Hi @minhduc0711, could you help review this PR? If all makes sense to you, could you close yours? Thanks!

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1196/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1196/

File: chapter_optimization/adadelta.md [hunk 1]
```diff
@@ -5,16 +5,35 @@ Adadelta is yet another variant of AdaGrad. The main difference lies in the fact
...
 $$\begin{aligned}
-    \mathbf{s}_t & = \rho \mathbf{s}_{t-1} + (1 - \rho) \mathbf{g}_t^2, \\
-    \mathbf{g}_t' & = \sqrt{\frac{\Delta\mathbf{x}_{t-1} + \epsilon}{\mathbf{s}_t + \epsilon}} \odot \mathbf{g}_t, \\
-    \mathbf{x}_t  & = \mathbf{x}_{t-1} - \mathbf{g}_t', \\
-    \Delta \mathbf{x}_t & = \rho \Delta\mathbf{x}_{t-1} + (1 - \rho) \mathbf{x}_t^2.
+    \mathbf{x}_t  & = \mathbf{x}_{t-1} - \mathbf{g}_t'. \\
 \end{aligned}$$
 
-The difference to before is that we perform updates with the rescaled gradient $\mathbf{g}_t'$ which is computed by taking the ratio between the average squared rate of change and the average second moment of the gradient. The use of $\mathbf{g}_t'$ is purely for notational convenience. In practice we can implement this algorithm without the need to use additional temporary space for $\mathbf{g}_t'$. As before $\epsilon$ is a parameter ensuring nontrivial numerical results, i.e., avoiding zero step size or infinite variance. Typically we set this to $\epsilon = 10^{-5}$. 
+So what is the rescaled gradient $\mathbf{g}_t'$? We can calculate as:
```
Comment (minhduc0711) [inline chapter_optimization/adadelta.md:22]:
```suggestion
So what is the rescaled gradient $\mathbf{g}_t'$? We can calculate it as follows:
```

File: chapter_optimization/adadelta.md [hunk 2]
```diff
@@ -5,16 +5,35 @@ Adadelta is yet another variant of AdaGrad. The main difference lies in the fact
...
+
+$$\begin{aligned}
+    \mathbf{g}_t' & = \frac{\sqrt{\Delta\mathbf{x}_{t-1} + \epsilon}}{\sqrt{{\mathbf{s}_t + \epsilon}}} \odot \mathbf{g}_t, \\
+\end{aligned}$$
+
+where the $\Delta \mathbf{x}_{t-1}$ is the leaky average of the squared rescaled gradients $\mathbf{g}_t'$. We set $\Delta \mathbf{x}_{t-1}$ to be $0$ at the start and update it every time with $\mathbf{g}_t'$, i.e.,
```
Comment (minhduc0711) [inline chapter_optimization/adadelta.md:28]:
```suggestion
where $\Delta \mathbf{x}_{t-1}$ is the leaky average of the squared rescaled gradients $\mathbf{g}_t'$. We initialize $\Delta \mathbf{x}_{0}$ to be $0$ and update it at each step with $\mathbf{g}_t'$, i.e.,
```

File: chapter_optimization/adadelta.md [hunk 3]
```diff
@@ -5,16 +5,35 @@ Adadelta is yet another variant of AdaGrad. The main difference lies in the fact
...
+
+$$\begin{aligned}
+    \Delta \mathbf{x}_t & = \rho \Delta\mathbf{x}_{t-1} + (1 - \rho) {\mathbf{g}_t'}^2,
+\end{aligned}$$
+
+and the $\epsilon$ is added to maintain the math stability, such as $10^{-5}$.
```
Comment (minhduc0711) [inline chapter_optimization/adadelta.md:34]:
```suggestion
and $\epsilon$ (a small value such as $10^{-5}$) is added to maintain numerical stability.
```

File: chapter_optimization/adadelta.md [hunk 4]
```diff
@@ -5,16 +5,35 @@ Adadelta is yet another variant of AdaGrad. The main difference lies in the fact
...
+In a nutshell Adadelta uses two state variables, $\mathbf{s}_t$ to store a leaky average of the second moment of the gradient and $\Delta\mathbf{x}_t$ to store a leaky average of the second moment of the change of parameters in the model itself. Note that we use the original notation and naming of the authors for compatibility with other publications and implementations (there is no other real reason why one should use different Greek variables to indicate a parameter serving the same purpose in momentum, Adagrad, RMSProp, and Adadelta). 
+
+Here is the technical details of Adadelta. Given the parameter du jour is $\rho$, we obtain the following leaky updates similarly to :numref:`sec_rmsprop`:
```
Comment (minhduc0711) [inline chapter_optimization/adadelta.md:10]:
```suggestion
Here are the technical details of Adadelta. Given the parameter du jour is $\rho$, we obtain the following leaky updates similarly to :numref:`sec_rmsprop`:
```

File: chapter_optimization/adadelta.md [hunk 5]
```diff
@@ -5,16 +5,35 @@ Adadelta is yet another variant of AdaGrad. The main difference lies in the fact
 
 ## The Algorithm
 
-In a nutshell Adadelta uses two state variables, $\mathbf{s}_t$ to store a leaky average of the second moment of the gradient and $\Delta\mathbf{x}_t$ to store a leaky average of the second moment of the change of parameters in the model itself. Note that we use the original notation and naming of the authors for compatibility with other publications and implementations (there is no other real reason why one should use different Greek variables to indicate a parameter serving the same purpose in momentum, Adagrad, RMSProp, and Adadelta). The parameter du jour is $\rho$. We obtain the following leaky updates:
+In a nutshell Adadelta uses two state variables, $\mathbf{s}_t$ to store a leaky average of the second moment of the gradient and $\Delta\mathbf{x}_t$ to store a leaky average of the second moment of the change of parameters in the model itself. Note that we use the original notation and naming of the authors for compatibility with other publications and implementations (there is no other real reason why one should use different Greek variables to indicate a parameter serving the same purpose in momentum, Adagrad, RMSProp, and Adadelta).
```
Comment (minhduc0711) [inline chapter_optimization/adadelta.md:8]:
```suggestion
In a nutshell, Adadelta uses two state variables, $\mathbf{s}_t$ to store a leaky average of the second moment of the gradient and $\Delta\mathbf{x}_t$ to store a leaky average of the second moment of the change of parameters in the model itself. Note that we use the original notation and naming of the authors for compatibility with other publications and implementations (there is no other real reason why one should use different Greek variables to indicate a parameter serving the same purpose in momentum, Adagrad, RMSProp, and Adadelta). 
```

**Comment by goldmermaid** [comment]
Fantastic! Thanks @minhduc0711 :)

**Comment by mli** [comment]
Job d2l-en/PR-1196/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1196/

**Pull request merged**